pkg/controller/cell.go

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package controller
+
+import (
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/metrics/metric"
+)
+
+const (
+	controllerGroupMetrics = "controller-group-metrics"
+
+	// labelControllerGroupName is the label used
+	// to identify controller-specific metrics
+	labelControllerGroupName = "group_name"
+)
+
+var (
+	// GroupMetricEnabled is populated with the set of ControllerGroups for which metrics are enabled
+	groupMetricEnabled = map[string]bool{}
+
+	// GroupRuns is a Prometheus-compatible metric for Controller
+	// runs, labeled by completion status and Group name
+	GroupRuns = metrics.NoOpCounterVec
+)
+
+var Cell = cell.Module(
+	"controller",
+	"Controllers and Controller Lifecycle management",
+	cell.Config(Config{}),
+	cell.Metric(NewMetrics),
+	cell.Invoke(Init),
+)
+
+type Metrics struct {
+	ControllerGroupRuns metric.Vec[metric.Counter]
```

Comment:
While being here, it could make sense to extract also the other controller-related metrics, for consistency. That would make sure that they are also exposed by the clustermesh-apiserver/kvstoremesh.

Reply:
This cell can definitely be made more useful. I think we should leave that to a follow-up PR so as not to expand the scope of this one too much.